plumbing/format/packfile/decoder_test.go

```diff
@@ -1,4 +1,4 @@
-package packfile
+package packfile_test
```

Comment:
Then you will have the same complexity in the other end. Then before use it (in several lines) you need to put the switch (in the better of the cases) 


The package is public due to a circular dependency